polyphemus/lib/models/etl_config.rb

```diff
@@ -0,0 +1,128 @@
+class Polyphemus
+  class EtlConfig < Sequel::Model
+    RUN_ONCE=0
+    RUN_NEVER=-1
+    STATUS_COMPLETED='completed'
+    STATUS_ERROR='error'
+
+    def self.next_to_run
+      self.exclude(archived: true).where {
+        Sequel.|(
+          {run_interval: Polyphemus::EtlConfig::RUN_ONCE},
+          Sequel.&(
+            (run_interval > 0),
+            Sequel.|(
+              {ran_at: nil},
+              ((ran_at + Sequel.lit("? * interval '1 second'", :run_interval)) < DateTime.now)
+            )
+          )
+        )
+      }.first
+    end
+
+    plugin :timestamps, update_on_create: true
+    plugin :column_encryption do |enc|
+      enc.key 0, Polyphemus.instance.config(:etl_secrets_key)
+      enc.column :secrets, format: :json
+    end
+
+    def etl_job_class
+      Polyphemus::Job.from_name(etl)
+    end
+
+    def validate_secrets(secrets)
+      etl_job_class.validate_secrets(secrets)
+    end
+
+    def validate_params(params)
+      etl_job_class.validate_params(params)
+    end
+
+    def run!
+      update(ran_at: DateTime.now)
```

Comment:
Should this only be updated after successful runs? Or it's okay that a failure also updates the ran_at value?

Reply:
Do we notify folks of failed or succesful runs? How will people get feedback, other than refreshing the ETL page?

Reply:
Unfortunately refreshing is required. Ideas welcome to fix this.

Reply:
I've baked in "send message to slack" as the default notification for errors in the other ETLs....might be an option to consider, you might be able to even mention (`@`) the user who ran the job, so they don't get lost in the notification channel.